.github/workflows/regression-test-auto.yml

```diff
@@ -0,0 +1,76 @@
+name: Regression Tests - Agglayer
+'on':
+  pull_request:
+    types: [opened, synchronize]  # Trigger on new pull requests and when existing ones are synchronized
+    branches:
+      - main
+    inputs:
+      bake_time:
+        description: bake time (minutes)
+        required: false
+        default: 20
+jobs:
+  deploy_devnet:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Set up Docker
+        uses: docker/setup-buildx-action@v1
+      - name: Clone and build agglayer image locally
+        run: |
+          git clone https://github.com/0xPolygon/agglayer.git
+          cd agglayer
+          if [ "$GITHUB_EVENT_NAME" == "pull_request" ]; then
+              GITHUB_SHA=$(cat $GITHUB_EVENT_PATH | jq -r .pull_request.head.sha)
+          fi
+          git checkout $GITHUB_SHA
+          echo "Containerizing and testing commit: $GITHUB_SHA"
+          docker compose -f docker/docker-compose.yaml build --no-cache agglayer
+      - name: Install Foundry
+        uses: foundry-rs/foundry-toolchain@v1
+      - name: Clone internal kurtosis-cdk repo
+        run: |
+          git clone https://github.com/0xPolygon/kurtosis-cdk.git
+          cd kurtosis-cdk
+          git checkout dan/jit_containers
+      - name: Install kurtosis
+        run: |
+          echo "deb [trusted=yes] https://apt.fury.io/kurtosis-tech/ /" | sudo tee /etc/apt/sources.list.d/kurtosis.list
+          sudo apt update
+          sudo apt install kurtosis-cli
+          kurtosis analytics disable
+      - name: Deploy CDK devnet on local github runner
+        run: |
+          cd kurtosis-cdk
+          kurtosis run --enclave cdk-v1 --args-file params.yml .
```

Comment:
Should we use a local params file instead of relying on the version matrix in the kurtosis repo?

Reply:
This file defines the configuration of the entire CDK stack, not just the agglayer. In the context of e2e tests, I think it's interesting to use the kurtosis-cdk configuration file. It allows you to test the agglayer in contact with the latest versions of the other components and, what's more, you have one less file to maintain. But up to you :)